src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { Options } from './types/Options';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [chosenUserId, setChosenUserId] = useState<number>(0);
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [isShown, setIsShown] = useState(false);
+  const [isModalLoading, setIsModalLoading] = useState(false);
+  const [option, setOption] = useState<Options>(Options.All);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos().then(data => {
+      setLoading(false);
+
+      let filteredData = data;
+
+      filteredData = filteredData.filter(filterTodo =>
+        filterTodo.title.toLowerCase().includes(query.toLowerCase()),
+      );
+
+      if (option === Options.Active) {
+        filteredData = filteredData.filter(checkTodo => !checkTodo.completed);
+      }
+
+      if (option === Options.Completed) {
+        filteredData = filteredData.filter(checkTodo => checkTodo.completed);
+      }
+
+      setTodos(filteredData);
```

Comment:
it is redundant to make filtering in use effect.
All what should do useEffect - it is fetch todos and set its in state.